src/ape_plugins/_cli.py

```diff
@@ -193,8 +243,9 @@ def install(cli_ctx, skip_confirmation):
             # NOTE: Be *extremely careful* with this command, as it modifies the user's
             #       installed packages, to potentially catastrophic results
             # NOTE: This is not abstracted into another function *on purpose*
-
-            args = [sys.executable, "-m", "pip", "install", "--quiet", package_name]
+            if upgrade:
```

Comment:
I am not sure it really matters to include `--upgrade` here because the plugin is being installed for the first time and it will install the latest

Reply:
I am thinking if the user puts in a specific version number and it installs. 

What happens when they run a different script to upgrade? Is that possible?

Reply:
I just tried it on `pip`. I looks like the `--upgrade` flag gets ignored. I think it an error would more helpful though.

That is probably out of scope for this PR, but we can make the options incompatible at the CLI-option level. Do want to  add that to the grocery list? I know a trick for this.

Reply:
Like `--upgrade` should be allowed with `--version` because they are incompatible
but that can be later on